sui_programmability/framework/src/natives/test_scenario.rs

```diff
@@ -177,82 +181,42 @@ fn get_inventory_for(
         .collect())
 }
 
-/// Return the ID's of objects deleted since a given `tx_begin_idx`
-pub fn deleted_object_ids(
+pub fn emit_wrapped_object_events(
     context: &mut NativeContext,
-    ty_args: Vec<Type>,
+    mut ty_args: Vec<Type>,
     mut args: VecDeque<Value>,
 ) -> PartialVMResult<NativeResult> {
-    debug_assert!(ty_args.is_empty());
-    debug_assert_eq!(args.len(), 1);
+    debug_assert_eq!(ty_args.len(), 1);
+    debug_assert_eq!(args.len(), 2);
 
+    let id_type = ty_args.pop().unwrap();
+    let removed = pop_arg!(args, VectorRef);
     let tx_begin_idx = pop_arg!(args, u64) as usize;
 
-    let deleted_ids: Vec<Value> = context
-        .events()
+    let mut removed_ids: HashSet<ObjectID> = HashSet::new();
+    for i in 0..removed.len(&id_type)?.value_as::<u64>()? {
+        let id = removed.borrow_elem(i as usize, &id_type)?;
+        let id_bytes = get_nested_struct_field(id.value_as::<StructRef>()?.read_ref()?, &[0])?;
+        removed_ids.insert(id_bytes.value_as::<AccountAddress>()?.into());
+    }
+
+    let processed_ids: HashSet<_> = context.events()[tx_begin_idx..]
         .iter()
-        .skip(tx_begin_idx)
         .filter_map(|(_, event_type_byte, _, _, val)| {
-            if *event_type_byte == EventType::DeleteObjectID as u64 {
-                Some(Value::vector_u8(get_deleted_id_bytes(val).to_vec()))
-            } else {
-                None
-            }
+            get_object_id_from_event(*event_type_byte, val)
         })
         .collect();
+    // Any object that was removed (and not returned) during the current transaction,
+    // but did not appear in any of the events, must be wrapped.
+    for id in removed_ids.difference(&processed_ids) {
```

Comment:
Will this iteration be deterministic (don't know what `HashSet.difference` guarantees)? Might want to use `BTreeSet`'s if not.

Reply:
Curious, is there a reason that determinism in unit tests matter?

Reply:
Just that general feeling nondeterminism in tests is a recipe for flakiness. Seems very unlikely to bite us here, I just prefer to avoid it because the future is uncertain.